examples/src/main/java/com/mapbox/navigation/examples/core/FreeDriveNavigationActivity.kt

```diff
@@ -0,0 +1,237 @@
+import com.mapbox.android.core.location.LocationEngineRequest
+import com.mapbox.android.core.location.LocationEngineResult
+import com.mapbox.api.directions.v5.models.DirectionsRoute
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.mapboxsdk.camera.CameraUpdateFactory
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.location.ReplayRouteLocationEngine
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.utils.Utils
+import com.mapbox.navigation.ui.map.NavigationMapboxMap
+import com.mapbox.navigation.ui.map.NavigationMapboxMapInstanceState
+import kotlinx.android.synthetic.main.activity_trip_service.mapView
+import kotlinx.android.synthetic.main.free_drive_navigation_layout.*
+import timber.log.Timber
+
+/**
+ * To ensure proper functioning of this example make sure your Location is turned on.
+ */
+class FreeDriveNavigationActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    companion object {
+        const val MAP_INSTANCE_STATE_KEY = "navgation_mapbox_map_instance_state"
+        const val DEFAULT_INTERVAL_IN_MILLISECONDS = 1000L
+        const val DEFAULT_MAX_WAIT_TIME = DEFAULT_INTERVAL_IN_MILLISECONDS * 5
+    }
+
+    private var mapboxNavigation: MapboxNavigation? = null
+    private var navigationMapboxMap: NavigationMapboxMap? = null
+    private var mapInstanceState: NavigationMapboxMapInstanceState? = null
+
+    @SuppressLint("MissingPermission")
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.free_drive_navigation_layout)
+        mapView.onCreate(savedInstanceState)
+        mapView.getMapAsync(this)
+
+        val mapboxNavigationOptions = MapboxNavigation.defaultNavigationOptions(
+                this,
+                Utils.getMapboxAccessToken(this)
+        )
+
+        mapboxNavigation = MapboxNavigation(
+                applicationContext,
+                Utils.getMapboxAccessToken(this),
+                mapboxNavigationOptions,
+                locationEngine = getLocationEngine()
+        ).also {
+            it.registerRouteProgressObserver(routeProgressObserver)
+        }
+
+        initListeners()
+    }
+
+    override fun onMapReady(mapboxMap: MapboxMap) {
+        mapboxMap.setStyle(Style.MAPBOX_STREETS) { style ->
+            navigationMapboxMap = NavigationMapboxMap(mapView, mapboxMap)
+            mapboxMap.moveCamera(CameraUpdateFactory.zoomTo(15.0))
+            navigationMapboxMap?.initializeLocationComponent()
```

Comment:
Couldn't this stay automatic? Would we like to run `NavigationMapboxMap` without a puck?

Reply:
The original constructor was doing too much. One side effect was the map would often rotate unexpectedly before navigation had even begun. This is an effort to decouple the instantiation of the object from initializing. In this example one happens right after the next but it wouldn't be necessary.  I'll look at doing something more automatic.